### PR TITLE
[datareposink] Create datareposink initial element @open sesame 04/13 22:50

### PR DESCRIPTION
**Pull request by songgot**
- Create datareposink initial version
- Add location property to set the path for saving a file
- Add basic file writing functionality

**Self evaluation:**
1. Build test: [*]Passed [ ]Failed [ ]Skipped
2. Run test: [*]Passed [ ]Failed [ ]Skipped

### CODE REVIEW

File: gst/datarepo/gstdatareposink.c [hunk 1]
```diff
@@ -0,0 +1,466 @@
...
+  GST_INFO_OBJECT (sink, "set caps %" GST_PTR_FORMAT, caps);
+
+  return TRUE;
+}
+
+/**
+ * @brief Change state of datareposink.
+ */
+static GstStateChangeReturn
+gst_data_repo_sink_change_state (GstElement * element,
```
Comment (jaeyun-jung) [inline gst/datarepo/gstdatareposink.c:275]: If we do nothing while changing the state, we can remove this callback.
(Also, set-caps and get-caps would be unnecessary.)

Comment (songgot) [reply gst/datarepo/gstdatareposink.c:275]: Thank you.
set-caps and get-caps is added for #4052.
I will remove change_state and I will add it feature if necessary.

File: gst/datarepo/gstdatareposink.c [hunk 2]
```diff
@@ -0,0 +1,414 @@
...
+/**
+ * @brief Start datareposink
+ */
+static gboolean
+gst_data_repo_sink_start (GstBaseSink * basesink)
+{
+  GstDataRepoSink *sink;
+
+  sink = GST_DATA_REPO_SINK_CAST (basesink);
+  UNUSED (sink);
```
Comment (jaeyun-jung) [inline gst/datarepo/gstdatareposink.c:395]: unnecessary line.

**Review by jaeyun-jung — APPROVED** [review]
LGTM, plz remove unnecessary line before merging this commit.

**Comment by songgot** [comment]
okay, thank you.

**Review by anyj0527 — APPROVED** [review]
LGTM ✔

**Review by wooksong — APPROVED** [review]
Looks OK to me.

**Pull request merged**